lib/string.js

```diff
@@ -70,6 +70,10 @@ internals.String.prototype._base = function (value, state, options) {
         if (this._flags.trim) {
             value = value.trim();
         }
+
+        if (this._flags.transform) {
```

Comment:
As I stated in #586 I'm a fan of a general `transform` function but my use case required only the `replace` so that would be a cleaner solution for me. Ideally maybe both?
@Marsup thoughts?